tower-http/src/content_encoding.rs

```diff
@@ -111,34 +186,279 @@ impl Encoding {
 pub(crate) fn encodings(
     headers: &http::HeaderMap,
     supported_encoding: impl SupportedEncodings,
-) -> Vec<(Encoding, f32)> {
+) -> Vec<(Encoding, QValue)> {
     headers
         .get_all(http::header::ACCEPT_ENCODING)
         .iter()
         .filter_map(|hval| hval.to_str().ok())
-        .flat_map(|s| s.split(',').map(str::trim))
+        .flat_map(|s| s.split(','))
         .filter_map(|v| {
-            let mut v = v.splitn(2, ";q=");
+            let mut v = v.splitn(2, ';');
 
-            let encoding = match Encoding::parse(v.next().unwrap(), supported_encoding) {
+            let encoding = match Encoding::parse(v.next().unwrap().trim(), supported_encoding) {
                 Some(encoding) => encoding,
                 None => return None, // ignore unknown encodings
             };
 
             let qval = if let Some(qval) = v.next() {
-                let qval = match qval.parse::<f32>() {
-                    Ok(f) => f,
-                    Err(_) => return None,
-                };
-                if qval > 1.0 {
-                    return None; // q-values over 1 are unacceptable
+                if let Some(qval) = QValue::parse(qval.trim()) {
+                    qval
+                } else {
+                    return None;
                 }
-                qval
             } else {
-                1.0f32
+                QValue::one()
             };
 
             Some((encoding, qval))
         })
-        .collect::<Vec<(Encoding, f32)>>()
+        .collect::<Vec<(Encoding, QValue)>>()
+}
+
+#[cfg(all(test, feature = "compression-gzip", feature = "compression-br"))]
```

Comment:
Great that you added so many tests! It would be awesome to have some property testing or fuzzing for this as well but I won't block this pr because of that. I do however want to have some tests which involves all 3 encodings as well just to be on the extra safe side.